File: sources/nest-backend/src/job/dto/create-job.dto.ts
import { IsDate, IsNumber, IsString } from "class-validator";

import { ApiProperty } from "@nestjs/swagger";
import { IsEnum } from "class-validator";
import { TransactionType } from "src/transaction/transaction.service";

export enum RecurringType {
  DAILY = 'DAILY',
  WEEKLY = 'WEEKLY',
  MONTHLY = 'MONTHLY',
  YEARLY = 'YEARLY',
}

export class CreateJobDto {
  @ApiProperty({ example: 'Sample job' })
  @IsString()
  job_name: string;

  @ApiProperty({ example: 'Sample job description' })
  @IsString()
  job_description: string;

  @ApiProperty({ example: '3' })
  @IsNumber()
  tag_id: number;   

  @ApiProperty({ example: 'INCOME' })
  @IsEnum(TransactionType)
  transaction_type: TransactionType;

  @ApiProperty({ example: 1 })
  @IsNumber()
  account_id: number;

  @ApiProperty({ example: 1 })
  @IsNumber()
  user_id: number;

  @ApiProperty({ example: '2024-01-01' })
  @IsDate()
  schedule_date: Date;

  @ApiProperty({ example: 100 })
  @IsNumber()
  amount: number;

  @ApiProperty({ example: 'DAILY' })
  @IsString()
  @IsEnum(RecurringType)    
  recurring_type: RecurringType;


}

File: sources/win-ui-frontend/Fin-Manager-v2/DTO/CreateJobDto.cs
using System.Text.Json.Serialization;

namespace Fin_Manager_v2.DTO;

public class CreateJobDto
{
    [JsonPropertyName("job_name")]
    public string JobName { get; set; } = string.Empty;

    [JsonPropertyName("job_description")]
    public string Description { get; set; } = string.Empty;

    [JsonPropertyName("tag_id")]
    public int TagId { get; set; }

    [JsonPropertyName("account_id")]
    public int AccountId { get; set; }

    [JsonPropertyName("user_id")]
    public int UserId { get; set; }

    [JsonPropertyName("schedule_date")]
    public DateTime ScheduleDate { get; set; }

    [JsonPropertyName("amount")]
    public decimal Amount { get; set; }
 
    [JsonPropertyName("recurring_type")]
    public string RecurringType { get; set; } = string.Empty;

    [JsonPropertyName("transaction_type")]
    public string TransactionType { get; set; } = string.Empty;
} 